### ISSUE TITLE
Prediction function is not working

### ISSUE THREAD

**Issue by Pentaonia** [vdorie/stan4bart#9]
I trained a model and want to predict that to a data frame and to a data frame out of a raster. Both is not working.

```
model.stan <- stan4bart(
    formula = pa ~ bart(. - road_distance - urban_distance - water_distance) +
    (1 | road_distance) + (1 | urban_distance) + (1 | water_distance),
    verbose = 0, 
    data = PA_train,
    #test = PA_test,
    weights = my_weights,
    chains = 1, 
    iter = 10, 
    bart_args = list(n.trees = 10, keepTrees = T)
)
model.stan.predict <- predict(model.stan, newdata = as.data.frame(EU_environment_low, xy = T), type = "ev")
```

Following error: 
```
Error in dimnames(indiv.bart) <- list(observation = NULL, sample = NULL,  : 
  length of 'dimnames' [3] must match that of 'dims' [2]
```
When I am using `type = "indiv.ranef"`, it is working but this is not what I need.

Hopefully there is a solution.

Best
Loubert

**Comment by Pentaonia** [comment]
Hi,
I was able to figure out what was wrong. As soon as I am able to fully test, I can provide you with the code snippet.

Best
Louis

**Comment by Pentaonia** [comment]
Hey, 

when you run the `stan4bart`-model with `chains = 1`, then the `indiv.bart` object in the `predict.stan4bartFit` does not fit to the 
`dimnames(indiv.bart) <-  list(observation = NULL, sample = NULL, chain = NULL)`. Therefore I included before that:

```
    if(length(dim(indiv.bart)) == 2){
    dim(indiv.bart) <- c(dim(indiv.bart),1)
    }
```

After including that, the prediction function is working properly, independent from the size of `chains` in the `stan4bart` call.

Best
Louis

**Comment by vdorie** [comment]
Thanks! Fixed in 6fc1a76.
